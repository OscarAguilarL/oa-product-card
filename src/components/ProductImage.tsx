import React, { useContext } from 'react';

import { productContext } from './ProductCard';

import noImage from '../assets/no-image.jpg';
import styles from '../styles/styles.module.css';

export interface Props {
    className?: string;
    img?: string;
    style?: React.CSSProperties;
    alt?: string;
}

export const ProductImage = ({ img, className, style, alt }: Props) => {
    const { product } = useContext(productContext);
    let imgToShow: string;

    if (img) {
        imgToShow = img;
    } else if (product.img) {
        imgToShow = product.img;
    } else {
        imgToShow = noImage;
    }

    return (
        <img
            src={imgToShow}
            className={`${styles.productImg} ${className}`}
            alt={product ? product.title : alt}
            style={style}
        />
    );
};
