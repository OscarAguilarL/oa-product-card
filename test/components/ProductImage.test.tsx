import * as React from 'react';
import renderer from 'react-test-renderer';

import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('Pruebas en el componente ProductImage', () => {
  test('debe de mostrar la imÃ¡gen enviada por argumento', () => {
    const wrapper = renderer.create(
      <>
        <ProductImage img="https://hola.jpg" alt="imagen de prueba" />
      </>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
