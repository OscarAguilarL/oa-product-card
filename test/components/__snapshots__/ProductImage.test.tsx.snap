// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ProductImage debe de mostrar la imagen del producto 1`] = `
<section
  className="productCard undefined"
>
  <img
    alt="Coffee Mug - Con Imagen"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</section>
`;

exports[`Pruebas en el componente ProductImage debe de mostrar la im√°gen enviada por argumento 1`] = `
<img
  alt="imagen de prueba"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
